<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="Lei Li leil@uchicago.edu" />

<meta name="date" content="2020-07-02" />

<title>VGenes, an integrated graphical tool for efficient, comprehensive and multimodal analyses of massive B-cell repertoire sequences</title>
<link rel="icon" type="image/x-icon" href="logo.ico"/>

<script src="http://ajax.aspnetcdn.com/ajax/jQuery/jquery-1.8.0.js"></script>

<link rel="stylesheet" type="text/css" href="css/style.css">


<script type="text/javascript"> 
function showorhide()  
{  
  var mychar = document.getElementById("con").style.display;
  if (mychar == "none") 
  {
    document.getElementById("con").style.display ="block";  
  }
  else
  {
    document.getElementById("con").style.display ="none";
  }
}
</script> 

<style>
.title {
  background: #A6CDE7;
  border-radius: 0px;
  display:block;
  width: 90%;
  padding: 12px;
  padding-left: 25px;
  font-size: 20px;
  font-weight: 700;
  font-size: 20px;
  border: 3px solid black;
}

h3 {
  padding-top: 10px;
}

</style>

<script type="text/javascript">
$(document).ready(function(){
  $('.title').click(function(){
    var father = $(this).parent('li');
    var display = father.children(".text").css('display');
    if(display == 'none'){
      father.children(".text").css('display','block');
      father.children(".img").css('display','block');
    } else {
      father.children(".text").css('display','none');
      father.children(".img").css('display','none');
    }
  });
});
</script>

</head>

<body>


<div class="navbox">
<ul class="nav">
  <li><a href="index.html">About</a></li>
  <li><a href="install.html">Install</a></li>
  <li><a class="active" href="tutorial.html">Tutorial</a></li>
  <li><a href="contact.html">Contact</a></li>
  <li><a href="index.html">Github Page</a></li>
</ul>
</div>

<div class="row">

 <div class="col s12 m3">
    <div class="table cover">
      <div class="cover-card table-cell table-middle">
        <div class="center">
          <img class="logo" src="img/logo.png">
        </div>
        
        <span class="author_name">VGenes, an integrated graphical platform for multimodal analyses of B and T cell receptor repertoire</span>
        <span class="author_bio">Wilson Lab</span>
        <span class="author_job">Weill Cornell Medicine</span>
      </div>
    </div>
  </div>

  <div class="col s12 m9">
    <div class="post-listing">
     <h1>Get started</h1>
     <ul>
       <li>Create a new BCR/TCR database or open our example database</li>
       <li>Import BCR/TCR sequences into your database</li>
       <li>Then enjoy VGenes to analyze and clone your sequences</li>
     </ul>
    <h1>Data import and annotation</h1>
    <ul>
      <li>
        <a class="title">Create VGenes database</a>
        <p class="text">On the welcome page, users are allowed to create new a VGenes database (DB) or open an existing VGenes DB. Users are allowed to check “backup my DB” to make a backup for the database being opened. This backup function is disabled by default for faster response. All recently visited VGenes DBs will be saved in a list, and users are allowed to quickly visit them using the “open recent” dropdown combo box. </p>
        <img class="img" src="img/fig1.png">
      </li>
      <li>
        <a class="title">Import data</a>
        <p class="text">New users can click “New VGene DB” to create their first VGenes database. After the VGenes DB is created, the program will ask users to import data into this database. Users will be directed to a data import dialog after they click “Yes”. Notably, users still have chance to import data into their database if they accidently clicked “No”. Users can click "data import icon" on the main interface to open the data import dialog. </p>

        <p class="text">As shown in the following figure, VGenes allows users to import raw data from multiple sources: 1) 10X single cell VDJ profiling generated by CellRanger, 2) FASTA files, 3) SEQ files. They are most commonly used data formats for either single-cell or bulk sequencing data in the field. VGenes supports both B cell receptor (BCR) and T cell receptor (TCR) data. Users can click the BCR and TCR icons to switch between two data types. </p>

        <img class="img" src="img/fig2.png" style="width: 50%">
        <p class="text">On the other hand, VGenes also allows users to import records from processed results: 1) IgBlast result (outfmt 3) , 2) IMGT results. Last, VGenes also allow users to import records from CSV files exported by VGenes, or VGene database themselves. It enables merging multiple VGene DBs. For example, for a large study with multiple subjects, users can create individual VGenes DB for each subjects, and then merge them into a big database using “import from VDB”.</p>

        <h3 class="text">Import from 10X VDJ</h3>
        <p class="text">Here, we’d like to demonstrate the data import process using 10X single cell VDJ profiling results (BCR data). First, users should click “Browse” button to browse and determine the path of 10X single cell VDJ profiling results folder. Then VGenes will automatically locate the two required files: consensus.fasta and filtered_contig_annotations.csv. VGenes will pops up an error message if those two files cannot be found under current folder. After that, users are allowed to setup a uniform format for sequence names, e.g. adding prefix that indicating sample origin and/or adding chain type indicators. Last, users are required to choose species, currently only Human and Mouse are supported. After click “OK”, VGenes will start to process the input data. A progress bar will be popped up to indicate the progress.  </p>
        <img class="img" src="img/fig3.png" style="width: 50%">

        <h3 class="text">Import from Fasta files</h3>
        <p class="text">Users are also allowed to import data from Fasta files. First, users should click “Browse” button to browse and determine the path of Fasta files. Users can either select multiple files at one time, or add multiple files one by one. Users are allowed to remove a file from current candidate list by a double click. The rest settings are as same as those in 10X VDJ page. BTW, file names will be used as prefix to all sequences to indicate sample origins and avoid sequence name redundancy. For example, sequences from file “../HA5.fasta” will have a prefix “HA5_”.   </p>
        <img class="img" src="img/fig4.png" style="width: 50%">

        <h3 class="text">Import from SEQ files</h3>
        <p class="text">Users are also allowed to import data from SEQ files. First, users should click “Browse” button to browse and determine the path of SEQ file folder. Users can either select multiple folders at one time, or add multiple folders one by one. Users are allowed to remove a file from current candidate list by a double click. The rest settings are as same as those in 10X VDJ page. BTW, folder names will be users as prefix to all sequences to indicate sample origins and avoid redundancy. For example, sequences from folder “../S266Cloning” will have a prefix “S266Cloning _”.  </p>
        <img class="img" src="img/fig5.png" style="width: 50%">

        <h3 class="text">Import from CSV files or VGene DBs</h3>
        <p class="text">Users are allowed to import records from CSV files or VGene DBs (VDB files). First, users should click “Browse” button to browse and determine the path of CSV files or VDB files. Users can either select multiple files at one time, or add multiple files one by one. Users are allowed to remove a file from current candidate list by a double click. Then just click “OK” to import. </p>
        <img class="img" src="img/fig6.png" style="width: 80%">

        <h3 class="text">Import from IgBlast or IMGT results</h3>
        <p class="text">Users are allowed to import records from IgBlast or IMGT results. Users just need to provide the result files then click “OK”.</p>
        <img class="img" src="img/fig7.png" style="width: 80%">

      </li>
      <li>
        <a class="title">Database annotation- link your VDJ with multi-omics</a>
        <p class="text">Users are allowed to annotate their VGenes databases with information from either multimodal profiling (e.g. transcriptome expression, clusters, antigen probe binding) or serological assays (e.g. antigen binding, neutralizing). This function connects V(D)J profiles with genotypic or phenotypic prosperities (especially for single cell data). </p>
        <img class="img" src="img/fig8.png" style="width: 50%;  box-shadow: none;">

        <p class="text">VGenes allows users to import annotation information from CSV files. The CSV file must contain a barcode column that indicate sequence identify. Users could open the annotation dialog by clicking the annotate icon on the toolbar on the main interface, or click Annotate->Import annotate from the menu. Users need to determine the “Anchor column” (barcode column) of the CSV file, and determine the barcode column in the VGenes DB. We pre-defined “Blank10 (10xBarcode)” of VGenes database as the barcode column. VGenes will annotate records by matching information from these columns. Then for all columns need to be imported, users can choose an existing column in VDB to annotate information in that column, or type a new column name to import information to a new column.  </p>
        <img class="img" src="img/fig9.png" style="width: 70%">

      </li>
    </ul>
    <h1>BCR/TCR analysis</h1>
      <ul>
        <li>
          <a class="title">Information display</a>
          <p class="text">VGenes integrates all information on a main UI with multiple tabs. </p>
          
          <h3 class="text">Basic information </h3>
          <p class="text">VGenes display basic information of sequences on the “Record” tab. As shown in the following figure, basic information includes: </br>
          • Sequence name </br>
          • Gene type </br>
          • V(D)J genes </br>
          • Isotype </br>
          • Barcodes </br>
          • Transcriptome clusters </br>
          • CDR3 details  </br>
          • Clone pool information </br>
          • Other details </br>
          </p>
          <img class="img" src="img/fig10.png" style="width: 70%">

          <h3 class="text">Sequence details</h3>
          <p class="text">VGenes display sequence details information on the “Sequence” tab. As shown in the following  figure, sequence details include:</br>
          • Sequence name</br>
          • Gene type</br>
          • V(D)J genes</br>
          • Isotype</br>
          • DNA sequence</br>
          • Amino acid sequence</br>
          • DNA germline sequence</br>
          • Amino acid germline sequence</br>
          In addition to information display, this tab also allows users to click features (e.g.g V gene, D gene, J gene) to select sequence.
          </p>
          <img class="img" src="img/fig11.png" style="width: 70%">

          <h3 class="text">IgBlast alignment</h3>
          <p class="text">VGenes display IgBlast alignment for current sequence on the “IgBlast Alignment” tab. This page reveals all the details of V(D)J alignment, SHMs and V, D, J locus information. </p>
          <img class="img" src="img/fig12.png" style="width: 70%">

          <h3 class="text">Records grid view</h3>
          <p class="text">VGenes displays all the records in a grid viewer on the “Database” tab. Users can browse more detailed information of their Ig genes. Users’ selection on the table viewer will be automatically synchronized to the tree viewer to the left, and vice versa. </p>
          <img class="img" src="img/fig13.png" style="width: 70%">
          <p class="text">We also integrated lots of functions in this tab page to facilitate data management.  </p>
          <img class="img" src="img/fig14.png" style="width: 50%; box-shadow: none;">

          <h3 class="text">Records table view</h3>
          <p class="text">VGenes displays all details of current record in a table viewer on the “Table” tab. By checking the “Save my edits automatically” box, VGenes allows users to edit their records.</p>
          <img class="img" src="img/fig15.png" style="width: 70%">

        </li>
        <li>
          <a class="title">Records edit</a>
          <p class="text">UUsers are allowed to edit their records in three ways: edit in table viewer, bulk edit, and advanced edit.  </p>
          
          <h3 class="text">Basic edit </h3>
          <p class="text">Users can edit records directly in the table viewer (double click to edit, not recommended), or use our bulk edit function. For bulk edit, users should check all records they want to edit first. Then they should set the column name to be edited, the old value and new value. At last click “Bulk Edit” to apply the edit.</p>
          <img class="img" src="img/fig16.png" style="width: 30%">

          <h3 class="text">Advanced edit </h3>
          <p class="text">Users are encouraged to edit their records using our advanced edit function. Users can active this function by clicking “Stat/Edit” on the “Record” and “Database” tabs. As shown in the following figures, for each existing value (or value range), users are allowed to update to a new value. Leaving new value blank means unchanged. Here we demonstrate the utility of this function by two examples: for non-numerical factors (e.g. Genetype, Isotype, clusters…), we update all isotypes into two broad categories: IgG/A and IgM/D; for numerical factors (e.g. number of mutations), we update values in different ranges into different levels. This function aims to help users to better and easier annotate their database.</p>
          <img class="img" src="img/fig17.png" style="width: 70%">

        </li>
        <li>
          <a class="title">Records search/match</a>
          <p class="text">It’s important for users to identify all records have a specify property. We designed multiple functions for users to select their interested records. </p>
          
          <h3 class="text">Basic search </h3>
          <p class="text">On the left-bottom corner, users can search records with given property. Users can simply select the search field (column), and type the target value, then click “Search” to search. All qualified records will be checked automatically. </p>
          <img class="img" src="img/fig18.png" style="width: 30%">
          <p class="text">To facilitate users’ operation, we also designed auto-complete functions for the search text field. As shown in below, users are allowed to type part of information and all matched values of current field will be displayed.  </p>
          <img class="img" src="img/fig19.png" style="width: 30%">
          
          <h3 class="text">Record match</h3>
          <p class="text">We developed a function for users to quickly locate all records that share same property with current sequence. Tow inputs for “Match” function will be automatically updated when users click most of fields on the “Record” tab. Then all qualified records will be checked on the tree viewer when users click “Match”. </p>
          <img class="img" src="img/fig20.png" style="width: 30%">

          <h3 class="text">Heavy chain (HC) and light chain (LC) pairing for single-cell data</h3>
          <p class="text">Furthermore, we also provided a function for users to quickly locate all sequences that shares the same barcode with their selected records. This function allows users to quickly find the paired heavy/light chains. Users can click the pairing icon in the toolbar to match paired heavy/light chains for current checked records. If there is not records have been checked, this function will match paired records for current mouse pick (current record displayed in “record” tab). This function will check all qualified records and return a detailed list with how many paired records have been matched for each query record (as shown in the following figure).</p>
          <img class="img" src="img/fig21.png" style="width: 50%">
          <p class="text">VGenes also provides a function for users to identify paired HC/LCs and export them into a new database. Users can click the following icon in the toolbar to active this function.</p>
          <img class="img" src="img/fig22.png" style="width: 10% ;box-shadow: none;">

          <h3 class="text">Advanced search/match</h3>
          <p class="text">Like we did for “advanced edit”, we developed advanced search/match function in VGenes too. Users are encouraged to search their records using our advanced search/match function. Users can active this function by clicking “Stat/Select” on the left-bottom corner of main UI. As shown in the following figures, users can check the checkboxes for values they want (left panel). Furthermore, for numerical field, for example number of mutations, users can click “this column is numerical” to display the distribution on a histogram, determine cutoffs to divide data into multiple ranges, and then check the ranges they want (right panel). Then users can click “Update checks” to update checks. As users may noticed, there are two different focuses on the top: “All records” and “Current checked records”. Users can focus on current check records to select records from current checks. This enables users to select their interested records using multiple filters.</p>
          <img class="img" src="img/fig23.png" style="width: 70%">

          <h3 class="text">Update search on graphical statistic</h3>
          <p class="text">Users are also allowed to update selections on graphical statistic (pie chart, bar chart, box chart, scatter chart, VDJ combination heatmap). To enable this function, users should check the “Enable update Selection” box on “GraphicalStat” tab. Notably, this function is based on current selection by default, it will clear all checks if there is no qualified record from current checks. For example, users can click “IgM” on a pie chart to check all IgM records. Then users click “IgD” on the pie chart. Because there is no qualified record in current selections, the checked records will be reset. Then users should click “IgD” again to select all IgD records.</p>
          <img class="img" src="img/fig24.png" style="width: 70%">
        </li>
        <li>
          <a class="title">Graphical statistic</a>
          <p class="text">We developed variety of functions for graphical statistic. These functions allow users to better understand and investigate their data. </p>
          <img class="img" src="img/fig25.png" style="width: 50%;box-shadow: none;">

          <h3 class="text">Pie chart</h3>
          <img class="img" src="img/fig26.png" style="width: 50%;box-shadow: none;">

          <h3 class="text">Bar chart</h3>
          <img class="img" src="img/fig27.png" style="width: 50%;box-shadow: none;">
          <img class="img" src="img/fig28.png" style="width: 50%;box-shadow: none;">
          <img class="img" src="img/fig29.png" style="width: 50%;box-shadow: none;">

          <h3 class="text">Box chart</h3>
          <img class="img" src="img/fig30.png" style="width: 50%;box-shadow: none;">

          <h3 class="text">Word cloud</h3>
          <img class="img" src="img/fig31.png" style="width: 50%;box-shadow: none;">

          <h3 class="text">River chart</h3>
          <img class="img" src="img/fig32.png" style="width: 50%;box-shadow: none;">

          <h3 class="text">Tree map</h3>
          <img class="img" src="img/fig33.png" style="width: 50%;box-shadow: none;">

          <h3 class="text">Scatter chart</h3>
          <img class="img" src="img/fig34.png" style="width: 50%;box-shadow: none;">
          <img class="img" src="img/fig35.png" style="width: 50%;box-shadow: none;">
          <img class="img" src="img/fig36.png" style="width: 50%;box-shadow: none;">

          <h3 class="text">Heatmap</h3>
          <img class="img" src="img/fig37.png" style="width: 50%;box-shadow: none;">

          <h3 class="text">V(D)J combination heatmap</h3>
          <img class="img" src="img/fig38.png" style="width: 50%;box-shadow: none;">

          <h3 class="text">Sankey diagram</h3>
          <img class="img" src="img/fig39.png" style="width: 50%;box-shadow: none;">

          <h3 class="text">Sequence logo</h3>
          <img class="img" src="img/fig40.png" style="width: 50%;box-shadow: none;">
        </li>

        <li>
          <a class="title">Clonal analysis</a>
          <p class="text">VGenes helps users to identify B cell clones, perform phylogenic analysis and other clonal analysis.  </p>
          <img class="img" src="img/fig41.png" style="width: 30%;box-shadow: none;">

          <h3 class="text">Identify clones</h3>
          <p class="text">We developed function in VGenes to identify B cell clones. In brief, this clone classifier uses the following criteria to classify BCR sequences into different clones: members of a clone should share same IGHV and IGHJ gene, and their CDR3 gene should have same length. Users can active this function by clicking Tools->Find Clonal. Users can only check interested records, then the “Find clonal” functions will be only applied on those checked records.
          </p>
          <img class="img" src="img/fig42.png" style="width: 40%;box-shadow: none;">

          <h3 class="text">Clone details</h3>
          <p class="text">We also developed UI to display details of B cell clones. On “Clone” tab, all identified B cell clones are listed to the left, allowing users to select to show their details: clone members, V, D, J usage, CDR3 length, number of members, and gene types. Users are also allowed to do some graphical statistic within selected clone. Furthermore, users can click “check all” button to check all members of this clone for further analysis.</p>
          <img class="img" src="img/fig43.png" style="width: 70%">

          <h3 class="text">Identify potential public clones </h3>
          <p class="text">VGenes allows users to identify potential public clones by determining a field that contains subject/batch/dataset information. To active this function, users can click “Tools->IdentifyPublicClone” in menu, or just click “Find Public” button in “Clones” tab. As shown in the following figure, all potential “public clones” (clones that across multiple subjects/batches/datasets) were highlighted in red so that users can further check the details.</p>
          <img class="img" src="img/fig44.png" style="width: 70%">

          <h3 class="text">Ig phylogeny for clones </h3>
          <p class="text">After B cell clones were identified, users are allowed to perform phylogenetic analysis for their interested clones. VGenes builds Ig phylogeny for selected clones using RAxML, a maximal likely-hood analysis tool (https://cme.h-its.org/exelixis/web/software/raxml/). </p>
          <img class="img" src="img/fig45.png" style="width: 70%">
          <p class="text">We also noticed that there are some Ig phylogeny specific tools available in research field, for example IgPhyML (http://evolve.zoo.ox.ac.uk/Evolve/IgPhyML.html). We developed functions to parse and visualize results from IgPhyML. Users should switch to “Load IgPhyML” tab under “Ig phylogeny” tab, then click “Load IgPhyML results” button to choose IgPhyML output file. Users also allowed to check “open on a pop-up window” to visualize trees on a pop-up window. </p>
          <img class="img" src="img/fig46.png" style="width: 70%">

          <h3 class="text">Sequence analysis for clones</h3>
          <p class="text">We developed powerful functions for users to further investigate sequences and alignments of clone members. Users can easily access sequence logos and multiple sequence alignment of all members of any specific clone with a few click.</p>
          <img class="img" src="img/fig47.png" style="width: 70%">

          <h3 class="text">Investigation of both HC and LC of clones </h3>
          <img class="img" src="img/fig48.png" style="width: 50%;box-shadow: none;">
        </li>

        <li>
          <a class="title">Antibody mutational analysis</a>
          <p class="text">VGenes allows users to perform mutational analysis for their sequences</p>
          
          <h3 class="text">Mutational load analysis</h3>
          <p class="text">VGenes counts SHMs separately for each region of V genes, and allows users to perform statistical analysis on SHM of these regions.</p>
          <img class="img" src="img/fig49.png" style="width: 70%">
          <p class="text">VGenes also can display all detail counts on a table.</p>
          <img class="img" src="img/fig50.png" style="width: 70%">

          <h3 class="text">Inferring SHM targeting models</h3>
          <p class="text">VGenes also infers SHM targeting models for selected BCR sequences. </p>
          <img class="img" src="img/fig51.png" style="width: 70%">
        </li>

        <li>
          <a class="title">Antibody selection and cloning</a>
          <p class="text">VGenes designed a variety of functions to help users to select qualified antibody candidates efficiently and effectively. For conventional BCR datasets, VGenes allows users to select their interested subsets using any criteria or combination of multiple criteria, and then investigate their V(D)J combinations, CDR3 diversities, and clonal expansions to determine qualified antibody candidates for further experimental characterization. This workflow is particularly powerful and efficient by taking advantage of novel multimodal BCR datasets with antigen probes.  </p>
          <p class="text">VGenes implemented a group of search and match functions designed for multimodal BCR data to facilitate users to focus on their interested subsets (e.g. antigen-specific B cells, B cells with high SHMs, naïve-like B cells) and explorer more about genetic preference and CDR3 diversities. Furthermore, we also developed an advanced graphical selector for users to apply multiple filtering criteria, allowing an effective selection of interested records from the multimodal datasets that have more than 100 different features. This selector is capable of visually selecting both numerical (numbers, e.g. number of SHMs, counts of antigen-probe binding, CDR3 length) and non-numerical (categories, factors, and levels, e.g. isotype, mRNA cluster, gene type) properties, allowing more flexibility and higher efficiency. </p>
          
          <h3 class="text">Antibody candidates selection</h3>
          <p class="text">By adding antigen-probe binding to the searching criteria, VGenes helps users to rapidly select those candidates that most likely to be having high affinity and neutralizing. For example, using a combination of criteria of NP negative + Spike positive + IgA/G in a COVID-19 BCR dataset to select antibody candidates.  </p>
          <img class="img" src="img/fig52.png" style="width: 70%;box-shadow: none;">

          <h3 class="text">HC and LC pairing and cleaning</h3>
          <p class="text">For single-cell datasets, after selected a subset of heavy chains, VGenes provides a function to help users to pair heavy chains (HC) with their light chains (LC) for the complete genome of antibodies. VGenes also helps users to easily check and filter out all improper B cell records (e.g. unpaired HC and LC, doublets, and multiplets) by investigating the number of HC and LC that share the same barcode. </p>
          <img class="img" src="img/fig53.png" style="width: 70%;box-shadow: none;">

          <h3 class="text">Sampling</h3>
          <p class="text">VGenes integrated three sampling methods for both bulk BCR data (sampling from BCR sequences, usually heavy chain) and single-cell BCR data (sampling from HC and LC pairs). VGenes allows users to determine all candidates by either checking all individual sequences or pairing HCs and LCs for complete antibody genome. After users determined candidates for sampling, three sampling methods, random sampling, stratified sampling, and representative sampling, are implemented for users to choose according to their requirement. In random sampling, VGenes randomly pick a user-specified number of samples. In stratified sampling, users are allowed to determine a group factor, then randomly sample from each group of the group factor. VGenes allows users to set a fixed or scaled sample size for each group. In representative sampling, we adopted an sampling algorithm from a previous study that can select representative samples by balancing multiple factors. This algorithm has three key steps: vectorizing, distancing, and clustering. Vectorizing step collects values of selected factors for each sample, and packs them into a vector. This step also normalize different factors to balance the contribution of each factor, avoiding the bias due to differences in magnitudes of measurements. The distancing step calculates pairwise numerical distances from the vectors. As there are two types of factors in VGenes, numerical factor and none-numerical factor, we designed different models to quantify differences of different factors. In brief, for numeric factors, the difference between two values is considered as their difference. For none-numeric factors, a binary system was applied: 0 indicates no difference and 1 indicates difference. Then the distance between any two sample will be calculated using a L2 norm (Euclidian distance) from the vector of difference. Using the pairwise distance matrix, clustering step clusters all samples into a user-specified number of clusters using K-medoids algorithm. Then VGenes considers all clusters centers as selected samples. VGenes also allows users to determine a few important factors that need to be comprehensively sampled (100% coverage requirement). For details about this representative sampling workflow please refer to a previous study. </p>
          <img class="img" src="img/fig54.png" style="width: 70%;box-shadow: none;">
          <img class="img" src="img/fig55.png" style="width: 70%">
          <p class="text">After sampling, users are allowed to click “Update candidate list from sampling results” (paired HC/LC mode) or “Select my sampling results in VGenes” (individual sequence mode) to apply sampling results to the candidate list or tree selection for further steps.</p>

          <h3 class="text">Antibody cloning</h3>
          <p class="text">For paired HC/LC antibodies, users are allowed to pair them and check the sequence in a candidate list table.  </p>
          <img class="img" src="img/fig56.png" style="width: 50%";box-shadow: none;>
          <p class="text">Users can choose “Sequence for Gibson Clone” from right bottom corner and click the button to the left to open the dialog for antibody sequences generation. This function will prioritizes paired HC/LC antibodies in the HC/LC page. If there is no sequences in the candidate list, this function will try to pull all checked sequences. </p>
          <p class="text">While sequences are loaded, VGenes checks all details of BCR sequences, including J-end sequence, translated sequences, and open reading frame, and highlights all issues with a red flag. Users are allowed to manually edit their sequences, and VGenes will automatically re-check users’ edits and turn the flag to green if the inspection passes.  </p>
          <p class="text">There are two common red flags: “Jend error” and “ORF error”. “Jend Errors” means the end of J gene is not correct, usually because of 1 or 2 extra nucleotides of the V(D)J sequence, users can just simply delete them. The translated AA, Jend info and the sequence check flag will be automatically updated. For Heavy chain, we define a 6-mer pattern “TCCTCA” as acceptable J-end and will highlight all different J-ends to notice users for a further check. For light chains, we define 6-mer patterns “TCGAAC”,  “ATTAAA”,  “ATCAAA” for kappa chain and define a 6-mer pattern “GTCCTA” for lambda chain as acceptable J-end. Notably, synonymous mutations were observed in the last six nucleotides of BCR genes, for example, “TCATCA” or “TCCTCG” instead of “TCCTCA” for heavy chains. For those cases, VGenes will highlight them and allows users to determine to whether accept or exclude them. Users can click "I think this sequence is OK" button on the top to make the flag green manually if they think the sequence is OK. “ORF error” means VGenes found something wrong in the translated AA sequence, so we suggest users check if there is any ambitious nucleotide in your sequence, or the beginning of VDJ sequence missed 1 or 2 nucleotides so that the ORF is wrong. The UI of Gibcon clone page is linked with the main VGenes page, therefore selections on main page will be automatically updated when users click a record on Gibson clone page. This feature is designed to facilitate users to check the sequence details for solving “ORF error” issue.  For “ORF error” issues, users can click and these records and investigate their IgBlast alignment on the main page to figure out if they need to fill the missed 1 or 2 (or delete the extra 1 or 2) nucleotides in the beginning of sequence to make sure the V(D)J sequence is correctly translated. </p>
          <p class="text">Users are also allowed to manually turn the flag of a specific sequence to green if they are confident with the current sequence. After the sequence inspection, users should provide Gibson upstream and downstream connector sequences. VGenes currently supports three expression vectors: Gibson AbVec, HT-AbVec, and AbVec classic. The connector sequences of these vectors are also customizable for better compatibility with users’ own cloning systems. Users can directly edit on the text editors to apply their own sequences for vectors. Users can click “Save” button to save the customized setting for later use. </p>
          <p class="text">After users click “Confirm”, VGenes will generated sequences for all records with indicator “Good” in a CSV file or FASTA file.  </p>
          <img class="img" src="img/fig57.png" style="width: 50%";box-shadow: none;>

          <p class="text">VGenes automatically added 5’ and 3’ connector sequences to all BCR sequences. All nucleotides of connector sequences are in lower case and all nucleotides of BCR sequences are in upper case. As shown in the following example:</p>

          <p class="text" style="font-family: Courier;">
            >HA5_48_H1</br>
            <b>atcctttttctagtagcaactgcaaccggtgtacattcc</b>CAGATCCAGCTGGTGCAGTCTGGGGCTGAG ... AGGGAACCCTGGTCACCGTCTCCTCA<b>gcgtcgaccaagggcccatcggtcttcc</b> 
          </p>
        </li>
      </ul>

      <h1>Export data</h1>
      <ul>
        <li>
          <a class="title">FASTA sequences</a>
          <p class="text">Users are allowed to export FASTA sequences for selected records. Users can export: </br>
          1)  nucleotide sequences </br>
          2)  amino acid sequences </br>
          3)  nucleotide germline sequences </br>
          4)  amino acid germline sequences </br>
          </p>
          <img class="img" src="img/fig58.png" style="width: 50%">
        </li>
        <li>
          <a class="title">AbVec cloning PCR report</a>
          <p class="text">Cloning report for AbVec cloning PCR primers for selected sequences.</p>
          <img class="img" src="img/fig58.png" style="width: 50%">
        </li>
        <li>
          <a class="title">Sequence summary report</a>
          <p class="text">A summarized report for selected sequences. It includs the following key information:Name, Project, Vgene, Dgene, Jgene, Vlocus, Jlocus, Productivity, IgBLASTmutationcount, CDR3DNA, CDR3peptide, CDR3length, CDR3isoelectricpoint, ClonalPool, Isotype, Sequence, SharedName. </p>
          <img class="img" src="img/fig58.png" style="width: 50%">
        </li>
        <li>
          <a class="title">Heavy/Light chain pairs report</a>
          <p class="text">This function will match paired heavy chain (HC) and light chain (LC) according to their barcode information for selected records. Only HC/LC pairs have 1 HC and 1 LC will be exported. HC/LC pairs have multiple HCs or LCs will be excluded. </p>
          <img class="img" src="img/fig58.png" style="width: 50%">
        </li>
        <li>
          <a class="title">CSV format full records</a>
          <p class="text">This function will export all detailed information for selected records in CSV file. </p>
          <img class="img" src="img/fig58.png" style="width: 50%">
        </li>
        <li>
          <a class="title">Amino Acid mutations report</a>
          <p class="text">This function will count and export number of amino acid mutations for selected sequences in CSV format. </p>
          <img class="img" src="img/fig59.png" style="width: 50%">
        </li>
        <li>
          <a class="title">Gibson Cloning sequence report</a>
          <p class="text">This function allows users to export their selected antibodies for Gibson cloning. VGenes developed a graphical UI to preview all selected sequences for users to confirm. VGenes automatically checks the end of sequences, and the ORF correction. All sequence passed checked will be highlighted in green with a “Good” indicator. Two error messages, “Jend error” and “ORF error”, are highlighted in red to attract users’ attention. Users are allowed to directly edit sequences in the text edit to fix the Jend error or ORF error. Indicators will be automatically updated when users edit the corresponding sequences. At last, VGenes allows users to make their own decision: users can click “I think this sequence is OK” button to compulsorily set indicator of a sequence to “Good” even to ignore all error messages. </br> After users confirmed all sequences, users should provide Gibson upstream and downstream connector sequences. After users click “Confirm”, VGenes will generated sequences for all records with indicator “Good” in a CSV file. </p>
          <img class="img" src="img/fig60.png" style="width: 50%">
        </li>
      </ul>

      
</div>
</div>
</div>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
